conf/utils.py

```diff
@@ -133,7 +133,12 @@ def get_process_affinity():
 
 
 def set_process_affinity(pid, cpus):
-    psutil.Process(pid).cpu_affinity(cpus)
+    try:
+        psutil.Process(pid).cpu_affinity(cpus)
+    except OSError as e:
+        # 22 = Invalid argument; PID has PF_NO_SETAFFINITY set
```

Comment:
Should I understand that we get `22` if and only if we have `PID has PF_NO_SETAFFINITY`? or are there other instances/flags where we can get error code `22`?

Reply:
As stated in [sched_setaffinity](https://man7.org/linux/man-pages/man2/sched_setaffinity.2.html), we would like to make it mandatory (from performance perspective).

@sureshmarikkannu @amarsri28, what are your thoughts on this?

Reply:
If we are not setting the affinity for some process (which has PF_NO_SETAFFINITY set) then it might be scheduled on the worker core right? In that case we do see severe performance hit for the data processing. In fact in one of the setup we allocated equal number of CPU's (same as configured no of workers) for the UPF pod and in that case we observed severe performance hit. So we always allocate number of workers + 1 CPU for the UPF pod to get best performance.

Reply:
We should at least fail with a meaningful error message then, rather than just a stack trace.

Reply:
Another option is lets keep this solution as it is and print **WARN** message when this kind of error happens. Something like "some process not affinitized, So it may cause performance impact".

Reply:
I much prefer that approach. I was initially going to put a WARN in the `except` clause, but was not sure if that was appropriate. I will update the patch if so